docs/synapse/userguides/storm_adv_control.rstorm

```diff
@@ -383,17 +544,85 @@ A **while loop** checks inbound nodes against a specified condition and performs
       <storm>
   }
 
-While loops are more frequently used for developer tasks, such as consuming from Queues; and are less common for day-to-day user use cases.
+While loops are more frequently used for developer tasks, such as consuming from Queues; and are less common for
+day-to-day user use cases.
+
+
+.. _flow-try-catch:
+
+Try...Catch Statement
++++++++++++++++++++++
+
+A **try...catch statement** allows you to attempt (try) a Storm operation and handle (catch) any errors if they
+occur. Because Storm's default behavior is to halt execution when an error occurs, try...catch statements allow
+for more graceful error handling within Storm. "Catching" an error allows the remainder of your Storm to
+continue executing.
+
+.. TIP::
+  
+  Storm supports some basic error handling (allowing you to "warn and continue" vs "error and halt") specifically
+  when creating nodes and setting properties or tags through the use of the :ref:`edit-try`.
+  
+**Syntax:**
+
+::
+  
+  try {
+      
+      <storm>
+      
+  } catch <name> as err {
+  
+      <storm>
+  }
+
+
+If the Storm in the try block runs without error, the catch block (or blocks) are ignored. If an error
+occurs, execution of the try block halts (any remaining Storm in the try block is ignored) and flow passes to
+the appropriate catch block to handle the error. Multiple catch blocks can be used to handle different kinds of
+errors.
+
+Because the catch block handles the error, any additional Storm (i.e., after the catch block) will continue to
+execute.
 
+In the catch block above, ``<name>`` can be the name of a single error type, a set of error types, or the asterisk
+( ``*`` ) to represent any error. When using multiple catch blocks, the asterisk can be used in the final block
+as a default case to catch any error not explicitly handled by a previous catch block.
+
+The catch block can return a status (e.g.,. ``return((1))``) or output a warning message (e.g., using ``$lib.warn()``
+- see :ref:`stormlibs-lib-warn`).
+
+**Example:**
+
+You have an "enrich" macro used to send various kinds of nodes to Storm commands that connect to third-party data
+sources. There is a particular data source that occasionally returns malformed data, which throws an error and
+causes the entire macro to halt. You want to isolate the Storm command for that vendor within a try...catch block
+so the macro will continue to run if an error is encountered.
+
+::
+  
+  try { 
+  
+      | enrich.badvendor
+      
+  } catch * as err {
+  
+     $lib.warn("BadVendor blew up again!")
+  }
```

Comment:
We will plumb out a more explicit example of throw and catching named exceptions; both from `throw` and SynErr primitives.

Reply:
Ty <3. I was proud of myself for getting this far. >D

Reply:
I've created a backlog story for us to get that additional language squared up; since the current write up is a great introduction.